### PR TITLE
table-level expectations; kl_divergence graphs in validation results

### PR DESCRIPTION
**Pull request by roblim**
- Add table-level expectations to validation page
- Add expected/observed distribution graphs to kl_divergence evrs in validation results page
- Minor docs tweaks in Data Docs section
- Fix broken Slack test

### CODE REVIEW

File: great_expectations/render/renderer/content_block/expectation_string.py [hunk 1]
```diff
@@ -298,6 +298,50 @@ def expect_multicolumn_values_to_be_unique(cls, expectation, styling=None, inclu
...
             }
         }]
+
+    @classmethod
+    def expect_table_column_count_to_equal(cls, expectation, styling=None, include_column_name=True):
+        params = substitute_none_for_missing(
+            expectation["kwargs"],
+            ["value"]
+        )
+
```
Comment (jcampbell) [inline great_expectations/render/renderer/content_block/expectation_string.py]:
```suggestion

```

File: great_expectations/render/renderer/content_block/expectation_string.py [hunk 2]
```diff
@@ -298,6 +298,50 @@ def expect_multicolumn_values_to_be_unique(cls, expectation, styling=None, inclu
...
+    @classmethod
+    def expect_table_column_count_to_equal(cls, expectation, styling=None, include_column_name=True):
+        params = substitute_none_for_missing(
+            expectation["kwargs"],
+            ["value"]
+        )
+    
+        template_str = "Must have exactly $value columns."
+
```
Comment (jcampbell) [inline great_expectations/render/renderer/content_block/expectation_string.py]:
```suggestion

```

File: great_expectations/render/renderer/content_block/expectation_string.py [hunk 3]
```diff
@@ -298,6 +298,50 @@ def expect_multicolumn_values_to_be_unique(cls, expectation, styling=None, inclu
...
+            }
+        }]
+
+    @classmethod
+    def expect_table_column_count_to_be_between(cls, expectation, styling=None, include_column_name=True):
+        params = substitute_none_for_missing(
+            expectation["kwargs"],
+            ["min_value", "max_value"]
+        )
+
```
Comment (jcampbell) [inline great_expectations/render/renderer/content_block/expectation_string.py]:
```suggestion

```

File: great_expectations/render/renderer/content_block/expectation_string.py [hunk 4]
```diff
@@ -298,6 +298,50 @@ def expect_multicolumn_values_to_be_unique(cls, expectation, styling=None, inclu
...
+        if params["min_value"] is None and params["max_value"] is None:
+            template_str = "May have any number of columns."
+        else:
+            if params["min_value"] is not None and params["max_value"] is not None:
+                template_str = "Must have between $min_value and $max_value columns."
+            elif params["min_value"] is None:
+                template_str = "Must have less than than $max_value columns."
+            elif params["max_value"] is None:
+                template_str = "Must have more than $min_value columns."
+
```
Comment (jcampbell) [inline great_expectations/render/renderer/content_block/expectation_string.py]:
```suggestion

```

File: great_expectations/render/renderer/column_section_renderer.py [hunk 5]
```diff
@@ -590,6 +590,9 @@ def __init__(self, table_renderer=None):
     def _render_header(cls, validation_results):
         column = cls._get_column_name(validation_results)
         
+        if not column:
+            column = "Table-Level Expectations"
+
```
Comment (jcampbell) [inline great_expectations/render/renderer/column_section_renderer.py:595]:
```suggestion

```

**Review by jcampbell — CHANGES_REQUESTED** [review]
Can you add a test for the existence of a table-level expectation in some rendered output?

I'm thinking along the following lines:
1. Build the site as we do now in output.
2. Open up an html file and just look for the strings "table-level expectations" and "row count"
3. Put said test in a new tests/render/renderer/test_page_renderer.py file

File: great_expectations/render/renderer/content_block/expectation_string.py [hunk 6]
```diff
@@ -1294,37 +1340,87 @@ def expect_column_kl_divergence_to_be_less_than(cls, expectation, styling=None,
             expectation["kwargs"],
             ["column", "partition_object", "threshold"]
         )
+        
+        expected_distribution = None
+        if not params.get("partition_object"):
+            template_str = "Column can match any distribution."
```
Comment (Aylr) [inline great_expectations/render/renderer/content_block/expectation_string.py:1343]: nice

File: tests/actions/test_validation_operators.py [hunk 7]
```diff
@@ -136,6 +136,7 @@ def test_errors_warnings_validation_operator_run_slack_query(basic_data_context_
         run_id="test_100"
     )
     slack_query = vo._build_slack_query(return_obj)
+    print(slack_query)
```
Comment (Aylr) [inline tests/actions/test_validation_operators.py]:
```suggestion
```

**Review by Aylr — APPROVED** [review]
lgtm I'm stoked about these little graphs!

**Pull request merged**